### DISCUSSION TITLE
Utilize existing assembler such as clang instead of nasm

### DISCUSSION THREAD

**Discussion by theoparis** [xiph/rav1e#3391] [Ideas]
I would like to remove the dependency on NASM by using my toolchain's clang binary, but rav1e's build.rs seems to only support nasm 😕  Is there something blocking this from being implemented?

Edit: Attempting to use the cc crate to compile the .asm files strangely results in:
![image](https://github.com/user-attachments/assets/d3ec9f41-90fe-49aa-845d-b8ff434ccd8f)

**Comment by lu-zero** [comment]
The x86_64 asm is written for nasm, a preprocessor or reimplementing the syntax nasm is using would be needed.

**Comment by mikerobi** [comment]
Rust now supports inline assembly, that may be a better option.

**Reply by negge** [reply]
See @tmatth comment below. We import these files as-is from [dav1d](https://code.videolan.org/videolan/dav1d) and we should not be in the business of converting 200k+ lines of assembly source code just to avoid a nasm dependency.

**Comment by tmatth** [comment]
Files like `src/x86/itx_avx2.asm` were imported from [dav1d](https://code.videolan.org/videolan/dav1d) and so I don't imagine we'd be changing toolchains in a way that would break that workflow (as we will continue to pull in asm as it gets improved/added to in `dav1d`).
